File: specification/storage/resource-manager/Microsoft.Storage/stable/2024-01-01/examples-java/TableServicesPut.java

import com.azure.resourcemanager.storage.fluent.models.TableServicePropertiesInner;
import com.azure.resourcemanager.storage.models.CorsRule;
import com.azure.resourcemanager.storage.models.CorsRuleAllowedMethodsItem;
import com.azure.resourcemanager.storage.models.CorsRules;
import java.util.Arrays;

/**
 * Samples for TableServices SetServiceProperties.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/storage/resource-manager/Microsoft.Storage/stable/2024-01-01/examples/TableServicesPut.json
     */
    /**
     * Sample code: TableServicesPut.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void tableServicesPut(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.storageAccounts().manager().serviceClient().getTableServices().setServicePropertiesWithResponse("res4410",
            "sto8607",
            new TableServicePropertiesInner().withCors(new CorsRules().withCorsRules(Arrays.asList(
                new CorsRule().withAllowedOrigins(Arrays.asList("http://www.contoso.com", "http://www.fabrikam.com"))
                    .withAllowedMethods(Arrays.asList(CorsRuleAllowedMethodsItem.GET, CorsRuleAllowedMethodsItem.HEAD,
                        CorsRuleAllowedMethodsItem.POST, CorsRuleAllowedMethodsItem.OPTIONS,
                        CorsRuleAllowedMethodsItem.MERGE, CorsRuleAllowedMethodsItem.PUT))
                    .withMaxAgeInSeconds(100).withExposedHeaders(Arrays.asList("x-ms-meta-*"))
                    .withAllowedHeaders(Arrays.asList("x-ms-meta-abc", "x-ms-meta-data*", "x-ms-meta-target*")),
                new CorsRule().withAllowedOrigins(Arrays.asList("*"))
                    .withAllowedMethods(Arrays.asList(CorsRuleAllowedMethodsItem.GET)).withMaxAgeInSeconds(2)
                    .withExposedHeaders(Arrays.asList("*")).withAllowedHeaders(Arrays.asList("*")),
                new CorsRule().withAllowedOrigins(Arrays.asList("http://www.abc23.com", "https://www.fabrikam.com/*"))
                    .withAllowedMethods(Arrays.asList(CorsRuleAllowedMethodsItem.GET, CorsRuleAllowedMethodsItem.PUT))
                    .withMaxAgeInSeconds(2000)
                    .withExposedHeaders(Arrays.asList("x-ms-meta-abc", "x-ms-meta-data*", "x-ms-meta-target*"))
                    .withAllowedHeaders(Arrays.asList("x-ms-meta-12345675754564*"))))),
            com.azure.core.util.Context.NONE);
    }
}

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-09-01/examples-js/TableServicesPut.js
const { StorageManagementClient } = require("@azure/arm-storage");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Sets the properties of a storage account’s Table service, including properties for Storage Analytics and CORS (Cross-Origin Resource Sharing) rules.
 *
 * @summary Sets the properties of a storage account’s Table service, including properties for Storage Analytics and CORS (Cross-Origin Resource Sharing) rules.
 * x-ms-original-file: specification/storage/resource-manager/Microsoft.Storage/stable/2021-09-01/examples/TableServicesPut.json
 */
async function tableServicesPut() {
  const subscriptionId = "{subscription-id}";
  const resourceGroupName = "res4410";
  const accountName = "sto8607";
  const parameters = {
    cors: {
      corsRules: [
        {
          allowedHeaders: ["x-ms-meta-abc", "x-ms-meta-data*", "x-ms-meta-target*"],
          allowedMethods: ["GET", "HEAD", "POST", "OPTIONS", "MERGE", "PUT"],
          allowedOrigins: ["http://www.contoso.com", "http://www.fabrikam.com"],
          exposedHeaders: ["x-ms-meta-*"],
          maxAgeInSeconds: 100,
        },
        {
          allowedHeaders: ["*"],
          allowedMethods: ["GET"],
          allowedOrigins: ["*"],
          exposedHeaders: ["*"],
          maxAgeInSeconds: 2,
        },
        {
          allowedHeaders: ["x-ms-meta-12345675754564*"],
          allowedMethods: ["GET", "PUT"],
          allowedOrigins: ["http://www.abc23.com", "https://www.fabrikam.com/*"],
          exposedHeaders: ["x-ms-meta-abc", "x-ms-meta-data*", "x-ms-meta-target*"],
          maxAgeInSeconds: 2000,
        },
      ],
    },
  };
  const credential = new DefaultAzureCredential();
  const client = new StorageManagementClient(credential, subscriptionId);
  const result = await client.tableServices.setServiceProperties(
    resourceGroupName,
    accountName,
    parameters
  );
  console.log(result);
}

tableServicesPut().catch(console.error);
